packages/loader/container-loader/src/container.ts

```diff
@@ -1046,7 +1063,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         [this._protocolHandler] = await Promise.all([protocolHandlerP, loadDetailsP]);
 
         const codeDetails = this.getCodeDetailsFromQuorum();
-        await this.loadContext(codeDetails, attributes, maybeSnapshotTree);
+        await this.loadContext(codeDetails, attributes, maybeSnapshotTree, undefined, pendingOps);
```

Comment:
I do not see any code that will ensure that we load from specific version / sequence number, to ensure that we could process local ops correctly (i.e. have same baseline as op requires).

Getting some thinking / prototyping in this space earlier would be very desired. Mostly because as of now, ODSP deletes snapshots pretty quickly from storage, keeping only last 10 snapshots. This causes troubles outside of that feature (delay-loading fails). We have active discussion to at least ensure that snapshot from which client loaded from stays available long enough to ensure correctness of offline & delay-loading scenarios. In other words, storage will track what snapshots we loaded from and keep them longer, while deleting quickly other snapshots (that we never used to load from).

Here, however, I assume we will want to load from some arbitrary snapshot (the last summary ack old container saw before it got disconnected / saved user state). This snapshot might not be around by the time we rehydrate new container. So we either need to raise it as blocking to ODSP and have discussion about it, or figure out if we can avoid it somehow, like maybe serializing more state, or loading from same snapshot this container loaded initially (i.e. going back in time further which slows down rehydration process substantially).

Reply:
Note that loader already has logic to process ops and return container at specific sequence number. I think this logic should move into Container, as it will be needed for this feature, but also for reloading context (versioning) - see https://github.com/microsoft/FluidFramework/issues/4685 where we might  need it too.